runtime/gc_vlhgc/CopyForwardScheme.hpp

```diff
@@ -118,9 +111,13 @@ class MM_CopyForwardScheme : public MM_BaseNonVirtual
 	MM_CopyScanCacheListVLHGC _cacheFreeList;  /**< Caches which are not bound to heap memory and available to be populated */
 	MM_CopyScanCacheListVLHGC *_cacheScanLists;  /**< An array of per-node caches which contains objects still to be scanned (1+node_count elements in array)*/
 	UDATA _scanCacheListSize;	/**< The number of entries in _cacheScanLists */
-	UDATA _scanCacheWaitCount;	/**< The number of threads currently sleeping on _scanCacheMonitor, awaiting scan cache work */
+	volatile UDATA _scanCacheWaitCount;	/**< The number of threads currently sleeping on _scanCacheMonitor, awaiting scan cache work */
 	omrthread_monitor_t _scanCacheMonitor;	/**< Used when waiting on work on any of the _cacheScanLists */
-	volatile UDATA _doneIndex;	/**< Incremented when _cacheScanLists are empty and we want all threads to fall out of _scanCacheMonitor */
+
+	volatile UDATA* _scanCacheWaitCountPtr;	/**< The number of threads currently sleeping on *_scanCacheMonitorPtr, awaiting scan cache work or work from packets*/
+	omrthread_monitor_t* _scanCacheMonitorPtr;	/**< Used when waiting on work on any of the _cacheScanLists or workPackets*/
+
```

Comment:
for the two pointers, since they would be used for both Scan Caches and WorkPackets, i'd give them more generic names _workQueue.... instead of _scanCache....

Reply:
done